feat: extend preview panel multi-selection with shared tag editing and update tests

JasonReb0rn opened the pull request:
### Summary
- Show shared tags for multiple selected entries and allow bulk tag removal
- Enable preview panel actions for multi-selection and adjust related tests
- Update preview file attributes text to indicate multi-selection and shared tags

<!--
^^^ Summarize the changes done and why they were done above.

By submitting this pull request, you certify that you have read the
[Contributing](https://docs.tagstud.io/contributing) page on our documentation site,
or in the project's [CONTRIBUTING.md](https://github.com/TagStudioDev/TagStudio/blob/main/docs/contributing.md) file.

IMPORTANT FOR FEATURES: Please verify that a feature request or some other form
of communication with maintainers was already conducted in terms of approving.

Thank you for your eagerness to contribute!
-->

### Tasks Completed

<!-- No requirements, just context for reviewers. -->

-   Platforms Tested:
    -   [X] Windows x86
    -   [ ] Windows ARM
    -   [ ] macOS x86
    -   [ ] macOS ARM
    -   [ ] Linux x86
    -   [ ] Linux ARM
    <!-- If an unspecified platform was tested, please add it here -->
-   Tested For:
    -   [X] Basic functionality
    -   [ ] PyInstaller executable <!-- Not necessarily required, but appreciated! -->
    <!-- If other important criteria was tested for, please add it here -->

JasonReb0rn committed:
Extend preview panel multi-selection with shared tag editing and update tests

TrigamDev commented:
Also, I think it would be nice if tags and fields that aren't shared between all selected entries are still displayed, but grayed out or low opacity

JasonReb0rn committed:
Fix preview panel race condition and add mixed tag display for multi-selection

JasonReb0rn commented:
> When you have an entry selected, select a second, large entry that takes a while to load, and then selecting a third entry before the second loads, the tags and fields will be stuck in a state where they don't update, even when selecting other entries. They will be the tags and fields of the very first entry.

Thank you for catching that. I believe I've addressed the race condition via timestamp-based cutoff.

Regarding the display of non-shared tags, I added a separate "mixed" tag section, which still allows for bulk tag removal. I think this approach is cleaner than graying tags out, but I'm happy to adjust if you prefer a different way forward.

Computerdores commented on src/tagstudio/qt/mixed/field_containers.py:
```diff
@@ -111,8 +111,87 @@ def update_from_entry(self, entry_id: int, update_badges: bool = True):
         self.cached_entries = [entry]
         self.update_granular(entry.tags, entry.fields, update_badges)
 
+    def update_from_entries(self, entry_ids: list[int], update_badges: bool = True):
```
In many other cases in the codebase functionality that can be done for either one or several items is implemented in a single method that checks whether a list or a single item was provided (or if the list only has one item). Doing the same for `update_from_entries` and `update_from_entry` would simplify the call in several places

Computerdores reviewed (commented):
Generally this is looking good code wise (haven't had time to test yet), but there is a couple of things that could be improved.

(I tried to be verbose in my explanations, but if anything is unclear let me know)

TrigamDev commented:
Adding a new field won't trigger the fields list to update. In order to see newly added fields, changing the selection is required.

TrigamDev commented:
This closes #808

JasonReb0rn committed:
fix bulk selection preview for partial tags and fields

JasonReb0rn commented:
Ok, I've spent some more time on this and made a broader pass over the bulk edit preview behavior.

The changes here mainly focus on how the tags and fields panel updates during multi-selection. Shared tags and fields are now separated more clearly from ones that only exist on some of the selected entries, the panel refresh behavior after bulk edits is more consistent, and the tag rendering logic was cleaned up a bit based on the review feedback.

JasonReb0rn committed:
improve tests

JasonReb0rn committed:
Merge origin/main into bulk-tag-editing, resolve field_containers conflict

JasonReb0rn committed:
Merge origin/main, resolve field_containers conflict
